src/Factories/FunFunFactory.php

```diff
@@ -1372,4 +1373,11 @@ public function setContentProvider( ContentProvider $contentProvider ): void {
 	private function getContentProvider(): ContentProvider {
 		return $this->pimple['content_provider'];
 	}
+
+	public function newMailTemplateFilenameTraversable(): \Traversable {
```

Comment:
What is returned is a very concrete `MailTemplateFilenameTraversable` which sure does implement `IteratgorAggregate` but may implement many more interfaces or add methods not described by any interface. We _can_ (as in having the privilege) be more specific about our **return** type w/o any disadvantage, while - sure - anyone hinting **parameter** types should hint the interface (s)he actually wants to use - Robustness principle.

For the use statements I thought there was a rule/agreement. If this is not the case... never mind. I like an inventory of used classes in the head of the file, also it's a small step from `\Foo` to `\Foo\Bar`.